### PR TITLE
Provide correct bound for the VRF check.

### PR DESCRIPTION
**Pull request by nc6**
Since in Praos the leader value doens't derive directly from the VRF
evaluation but from a hash, the bounds are different. This change (in
conjunction with a related change in the ledger) means that one has to
provide the bounds explicitly when comparing the leader value.

### CODE REVIEW

File: ouroboros-consensus-protocol/src/Ouroboros/Consensus/Protocol/Praos/VRF.hs [hunk 1]
```diff
@@ -100,9 +103,11 @@ vrfLeaderValue ::
   Crypto c =>
   proxy c ->
   CertifiedVRF (VRF c) InputVRF ->
-  Natural
-vrfLeaderValue p =
-  bytesToNatural . hashToBytes . hashVRF p SVRFLeader
+  BoundedNatural
+vrfLeaderValue p cvrf =
+  assertBoundedNatural
+    (8 * fromIntegral (sizeHash (Proxy @(HASH c))))
```
Comment (kderme) [inline ouroboros-consensus-protocol/src/Ouroboros/Consensus/Protocol/Praos/VRF.hs:109]: Shouldn't be 2 ^ this?

Comment (nc6) [reply ouroboros-consensus-protocol/src/Ouroboros/Consensus/Protocol/Praos/VRF.hs:109]: Yes, yes it should!

**Review by dcoutts — APPROVED** [review]
LGTM.

Including the referenced ledger commit to adjust `checkLeaderValue` to use `BoundedNatural`.

**Comment by nc6** [comment]
bors merge

**Comment by nc6** [comment]
bors merge

**Comment by nc6** [comment]
bors retry

**Comment by nc6** [comment]
bors retry

**Comment by nc6** [comment]
bors retry

**Pull request merged**